### DISCUSSION TITLE
Feature Request: Files hosted over HTTP without Range header support

### DISCUSSION THREAD

**Discussion by samansmink** [duckdb/duckdb#5541] [Ideas]
This is useful for servers/services that don't support range requests.

A notable example is the Google sheets csv export feature, which doesn't support range requests: https://developers.google.com/drive/api/guides/manage-downloads#partial_download

this would allow queries such as:
```sql
select * from read_csv_auto('https://docs.google.com/spreadsheets/d/1XGCy0tYU5YcEouO09_ErZIyqjA-VJ4pidLZmMmJkEdk/gviz/tq?tqx=out:csv&sheet=Sheet1&range=A1:C3')
```

Note that this feature already exists on DuckDB-WASM.

**Comment by Mytherin** [comment]
This should be relatively straightforward for the CSV reader as we can simply toggle the `CanSeek` option to false on the file, leading to the CSV reader using only sequential reads.

For Parquet files this will require actually downloading the file and storing it either on disk or in-memory.

**Reply by frafra** [reply]
Using `curl`+ `/dev/stdin` to avoid seeking is far from optional: it would be great to be able to ingest directly data from these kinds of endpoints.
`read_csv` fails even with `auto_detect=false`.

**Comment by djouallah** [comment]
@Mytherin if you implement some form of local disk cache, I will write songs about you 🥇 the speed of object store are really horrible ( not duckdb fault)
